MetaMask/metamask-mobile
# fix: fix signature requested analytics

## **Description**

This PR fixes an issue where the `Signature Requested` mobile analytics event was not being sent when a dapp sends a sign request. Only the `Signature Approved` analytics where being sent.

Examples:
From [MetaMask SDK Wagmi test dapp](https://mmsdk-wagmi-test-dapp.vercel.app) (mobile browser):
```
{"event": "Signature Requested", "properties": {"account_type": "MetaMask", "chain_id": "11155111", "dapp_host_name": "wagmi.io", "request_platform": "web-mobile", "request_source": "MetaMask-SDK-Remote-Conn", "signature_type": "personal_sign", "version": "N/A"}, "type": "track"}
```
From [WalletConnect test dapp](https://appkit-lab.reown.com) (mobile browser):
```
{"event": "Signature Requested", "properties": {"account_type": "MetaMask", "chain_id": "11155111", "dapp_host_name": "appkit-lab.reown.com", "request_platform": undefined, "request_source": "WalletConnect", "signature_type": "eth_signTypedData_v4", "version": "N/A"}, "type": "track"}
```

From iOS SDK test dapp:
```
{"event": "Signature Requested", "properties": {"account_type": "MetaMask", "chain_id": "11155111", "dapp_host_name": "dubdapp.com", "request_platform": "ios", "request_source": "MetaMask-SDK-Remote-Conn", "signature_type": "eth_signTypedData_v4", "version": "N/A"}, "type": "track"}
```

## **Related issues**

Fixes: [SDK-65](https://consensyssoftware.atlassian.net/browse/SDK-65)

## **Manual testing steps**

1. Go to this [mmsdk-wagmi-test-dapp](https://mmsdk-wagmi-test-dapp.vercel.app/) on mobile device
2. Connect to MetaMask
3. Sign message
4. Check the logs that `Signature Requested` is being logged to analytics and also check Segment that the event is being received.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**



[SDK-65]: https://consensyssoftware.atlassian.net/browse/SDK-65?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
